src/Routes/PrivateRoute.tsx

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react'
+import type { RouteProps } from 'react-router-dom'
+import { Redirect, Route, Switch } from 'react-router-dom'
+
+type PrivateRouteProps = {
+  hasWallet: boolean
+} & RouteProps
+
+export const PrivateRoute = ({ hasWallet, ...rest }: PrivateRouteProps) => {
+  const { location } = rest
+
+  const to = useMemo(
+    () => ({
+      pathname: '/connect-wallet',
+      search: `returnUrl=${location?.pathname ?? '/trade'}`,
+    }),
+    [location],
+  )
+
+  return (
+    <Switch location={location}>
+      <Route {...rest} path='/markets' />
+      <Route {...rest} path='/assets' />
+      <Route {...rest} path='/rfox' />
+      <Route {...rest} path='/explore' />
+      <Route {...rest} path='/pools' />
+      <Route {...rest} path='/earn' />
+      <Route {...rest} path='/buy-crypto' />
+      <Route {...rest} path='/assets' />
+      <Route {...rest} path='/flags' />
```

Comment:
Explicitly unprotected routes to be hit before the redirect. Since most users aren't really going to land in any of these except for markets for the time being (though could, and we handle things gracefully in all routes except lending), and lending is an empty, seemingly broken state, didn't add it back for the time being.
Would be automagically handled if/when we remove the concept of private routes again.